pkg/client/connector/intercept.go

```diff
@@ -154,11 +269,7 @@ func (tm *trafficManager) addIntercept(c context.Context, ir *rpc.CreateIntercep
 	}
 
 	<-tm.startup
-	intercepts, err := actions.ListMyIntercepts(c, tm.managerClient, tm.session().SessionId)
-	if err != nil {
-		return nil, err
-	}
-	for _, iCept := range intercepts {
+	for _, iCept := range tm.getCurrentIntercepts() {
```

Comment:
Just to check my understanding.. in the past we limited the intercepts we iterated through to be the ones associated with the session ID for the client. But since in the loop we are checking if the names are the same (which implicitly includes the sessionID) or that the targetPort + targetHost are already in use (implicitly checking for the client the call is coming from), we can replace this with iterating through _all_ the intercepts without issue?

Reply:
The `WatchInterecps()`, when called from a client, will only include those intercepts that are associated with that client. The traffic-manager ensures that all others are rejected [by this filter](https://github.com/telepresenceio/telepresence/blob/thallgren/intercept-survival/cmd/traffic/cmd/manager/service.go#L224-L227).